### PR TITLE
Episode of care measure support

### PR DESCRIPTION
**Pull request by mgramigna**
# Summary

Added support for episode of care measures, as well as additional output for number of episodes in each population in the CSV when dealing with an episode of care measure.

Bumped to a new major version `1.1.0` for release when this goes through.

## New behavior

Implemented proper handling of `List` return types from the CQL rather than assuming boolean. If the populations return a list, we have an episode of care measure, and this is automatically detected by the CLI (no additional flag necessary).

## Code changes

* CSV output rows can now differ if dealing with an episode of care measure
  * New columns for no. of episodes for each population
* Calculation logic now differs based on the resulting type returned from the population definitions in the CQL
* Added a small blurb in the README describing the differing output for episode of care measures
* Added a new test mocking a `List` returned from the CQL treating it as an episode of care measure.

# Testing guidance

`npm test` for the unit tests. For manual testing, there is a slight dependency on patients/proper environment for EXM104 since it is a WIP right now, but hopefully a visual review and proper unit test along with potential verification from @okeefm should be enough to push this through.

### CODE REVIEW

File: src/utils/calculation.js [hunk 1]
```diff
@@ -55,13 +61,28 @@ const calculate = async (url, cql, patientId, periodStart, periodEnd) => {
...
+        result[populationIdentifier] = value.valueString === 'true';
+      } else if (valueType.valueString === 'List') {
+        // For a List return type in episode of care measures, we want a non-zero
+        // amount of episodes for this patient
+        const isEmptyPopulation = value.valueString === '[]';
+        const numEpisodes = isEmptyPopulation ? 0 : value.resource.entry.length;
+        result[populationIdentifier] = numEpisodes > 0;
+
+        // Add a column for number of episodes in the population for episode of care measures
+        if (!result.counts) {
```
Comment (mayerm94) [inline src/utils/calculation.js]: Thoughts on adding result.ids = <ids of episodes in the value.resource.entry list> To make it easier to debug _which_ episodes are or are not satisfying the logic.

Comment (mgramigna) [reply src/utils/calculation.js]: Yeah I can look at this. What do you think the best format is for outputting these into csv? Another column for `epsiodeIds` that contains a list of resource ids? Any other suggestions?

Comment (mayerm94) [reply src/utils/calculation.js]: I think a column that contains a list of the resource ids would be perfect

**Comment by mgramigna** [comment]
I just pushed up a change that added columns to the csv for the list of resource IDs. The unit test now tests this as well.

I also took this opportunity to output more logging information in a meaningful way, as it was useful for helping me implement this. So the library now uses [winston](https://www.npmjs.com/package/winston) for logging, and the cli has a `-v/--verbose` flag to enable more debug logs (false by default). Hope this helps for the future.

**Comment by okeefm** [comment]
I'm getting a couple of test failures from `npm test`:

```
2.5.3 ➜  fhir-bundle-calculator git:(episode-of-care-support) ✗ npm test

> fhir-bundle-calculator@1.1.0 test /Users/mokeefe/code/fhir-bundle-calculator
> jest

 FAIL  test/calculation.test.js
  ✕ patient in the numerator should yield an object of all true (176ms)
  ✕ patient in ipp/denom but not numerator should yield proper booleans (119ms)
  ✕ illegal cql should yield no population results and an error message (126ms)
  ✕ episode of care measure should contain episode counts (130ms)

  ● patient in the numerator should yield an object of all true

    Request failed with status code 404

      at createError (node_modules/axios/lib/core/createError.js:16:15)
      at settle (node_modules/axios/lib/core/settle.js:17:12)
      at IncomingMessage.handleStreamEnd (node_modules/axios/lib/adapters/http.js:237:11)

  ● patient in ipp/denom but not numerator should yield proper booleans

    Request failed with status code 404

      at createError (node_modules/axios/lib/core/createError.js:16:15)
      at settle (node_modules/axios/lib/core/settle.js:17:12)
      at IncomingMessage.handleStreamEnd (node_modules/axios/lib/adapters/http.js:237:11)

  ● illegal cql should yield no population results and an error message

    Request failed with status code 404

      at createError (node_modules/axios/lib/core/createError.js:16:15)
      at settle (node_modules/axios/lib/core/settle.js:17:12)
      at IncomingMessage.handleStreamEnd (node_modules/axios/lib/adapters/http.js:237:11)

  ● episode of care measure should contain episode counts

    Request failed with status code 404

      at createError (node_modules/axios/lib/core/createError.js:16:15)
      at settle (node_modules/axios/lib/core/settle.js:17:12)
      at IncomingMessage.handleStreamEnd (node_modules/axios/lib/adapters/http.js:237:11)
```

**Comment by mgramigna** [comment]
@okeefm  This appeared to be an error with having the MITRE proxy variables set in your environment, as nock was not intercepting the requests properly since they were being tunneled through the proxy. I pushed up a workaround for this that takes advantage of localhost being a mock url and never being proxied regardless of the environment variables. They should pass for you now.

File: src/utils/calculation.js (hunk 1 above)
Comment (okeefm) [reply src/utils/calculation.js]: Reviewing the column, it looks good on CMS104.

**Pull request merged**